jazz_codeq/test/test.js

```diff
@@ -0,0 +1,757 @@
+			serviceContext = {};
+		})
+
+		it('should return report data when error code is 404 for getReportOnError', () => {
+			const err = {
+				report_error: 'report error',
+				code: 404
+			};
+			
+			const getReportStub = sinon.stub(utils, 'getReport').resolves({output: 'report data' });
+
+			index.getReportOnError(err, metrics, {}, serviceContext)
+			.then(result => {
+				sinon.assert.calledOnce(getReportStub);
+
+				expect(result.data.output).to.eq("report data");
+
+				getReportStub.restore();
+			});
+		});
+
+		it('should return report error when getReport rejects for getReportOnError', () => {
+			const err = {
+				report_error: 'report error',
+				code: 404
+			};
+			
+			const getReportStub = sinon.stub(utils, 'getReport').rejects({report_error: 'report error' });
+
+			index.getReportOnError(err, metrics, {}, serviceContext)
+			.then(result => {
+				sinon.assert.calledOnce(getReportStub);
+
+				expect(result.error).to.eq("report error");
+
+				getReportStub.restore();
+			});
+		});
+
+		it('should return quality error message when getReportOnError is called with decrypt error ', () => {
+			const err = {
+				decrypt_error: 'report error'
+			};
+
+			let result = index.getReportOnError(err, metrics, {}, serviceContext)
+
+			expect(result.error).to.eq("Error getting quality report from provider.");
+		});
+	});
+});
+
+describe('validation tests', () => {
+  	it('should return missing fields when not all are provided within the query', () => {
+		const requiredFields = ["domain", "service", "environment"];
+		const query = {"domain": "value"};
+		let validateMissingFields = validation.validateMissingFields(requiredFields, query);
+
+		expect(validateMissingFields).to.eql("service, environment");
+	});
+	  
+	it('should return no missing fields when all the fields are provided in the query', () => {
+		const requiredFields = ["domain", "service", "environment"];
+		const query = { "domain": "value", "service": "value", "environment": "value" };
+		let validateMissingFields = validation.validateMissingFields(requiredFields, query);
+
+		expect(validateMissingFields).to.eql("");
+	});
+	  
+	it('should return valid from date using moment', () => {
+		const from = '2018-05-11';
+		let validFrom = validation.validateFromDate(from);
+
+		expect(validFrom).to.eql("2018-05-11T12:00:00-0000");
+	});
+	
+	it('should return default date when from is not provided', () => {
+		const validFrom = validation.validateFromDate();
+		const DATE_FORMAT = 'YYYY-MM-DDThh:mm:ss';
+		const APPEND_ZEROES = '-0000';
+		let expectedResult = moment().subtract(1, 'days').format(DATE_FORMAT) + APPEND_ZEROES;
+
+		expect(validFrom).to.eql(expectedResult);
+	});
+	  
+	it('should return valid to date using moment', () => {
+		const to = '2018-05-11';
+		let validFrom = validation.validateToDate(to);
+
+		expect(validFrom).to.eql("2018-05-11T12:00:00-0000");
+	});
+	
+	it('should return default to date when to date is not provided', () => {

```

Comment:
How about failure case for `validateToDate` (date is not in valid format)?

Reply:
added test